java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/core/DbzCdcEventConsumer.java

```diff
@@ -76,6 +83,22 @@ public void handleBatch(
             DebeziumEngine.RecordCommitter<ChangeEvent<SourceRecord, SourceRecord>> committer)
             throws InterruptedException {
         var respBuilder = GetEventStreamResponse.newBuilder();
+        // wake up the thread blocked in `DbzCdcEngineRunner::start()`
+        startSignal.countDown();
```

Comment:
Do we need to `countDown` on each `handleBatch` invocation? Should we put it in L89 only?

Reply:
After review the code, I found that we can eliminate the `startSignal`, since the source executor actually synchronized on the **handshake** message. Only after the split reader recved the handshake message, it will return from its `new` function, then source will forward the initial barrier to downstream.
https://github.com/risingwavelabs/risingwave/blob/40c020f75428648b2bb9130a035c5abba02fa4ab/src/stream/src/executor/source/source_executor.rs#L410-L413